Which exception is thrown by read() method?